crates/uv-toolchain/src/platform.rs

```diff
@@ -1,152 +1,127 @@
-            "openbsd" => Ok(Self::OpenBsd),
-            "dragonfly" => Ok(Self::Dragonfly),
-            "illumos" => Ok(Self::Illumos),
-            "haiku" => Ok(Self::Haiku),
-            _ => Err(Error::OsNotSupported(s.to_string())),
+impl Display for Libc {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Some(env) => write!(f, "{env}"),
+            Self::None => write!(f, "none"),
         }
     }
 }
 
-impl fmt::Display for Arch {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Self::Aarch64 => write!(f, "aarch64"),
-            Self::Armv6L => write!(f, "armv6l"),
-            Self::Armv7L => write!(f, "armv7l"),
-            Self::Powerpc64Le => write!(f, "ppc64le"),
-            Self::Powerpc64 => write!(f, "ppc64"),
-            Self::X86 => write!(f, "i686"),
-            Self::X86_64 => write!(f, "x86_64"),
-            Self::S390X => write!(f, "s390x"),
+impl Display for Os {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &**self {
+            target_lexicon::OperatingSystem::Darwin => write!(f, "macos"),
+            inner => write!(f, "{inner}"),
         }
     }
 }
 
-impl FromStr for Arch {
+impl Display for Arch {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match &**self {
+            target_lexicon::Architecture::X86_32(target_lexicon::X86_32Architecture::I686) => {
+                write!(f, "x86")
+            }
+            inner => write!(f, "{inner}"),
+        }
+    }
+}
+
+impl FromStr for Os {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s.to_lowercase().as_str() {
-            "aarch64" | "arm64" => Ok(Self::Aarch64),
-            "armv6l" => Ok(Self::Armv6L),
-            "armv7l" | "arm" => Ok(Self::Armv7L),
-            "powerpc64le" | "ppc64le" => Ok(Self::Powerpc64Le),
-            "powerpc64" | "ppc64" => Ok(Self::Powerpc64),
-            "x86" | "i686" | "i386" => Ok(Self::X86),
-            "x86_64" | "amd64" => Ok(Self::X86_64),
-            "s390x" => Ok(Self::S390X),
-            _ => Err(Error::ArchNotSupported(s.to_string())),
+        let inner = match s {
+            "macos" => target_lexicon::OperatingSystem::Darwin,
+            _ => target_lexicon::OperatingSystem::from_str(s)
+                .map_err(|()| Error::UnknownOs(s.to_string()))?,
+        };
+        if matches!(inner, target_lexicon::OperatingSystem::Unknown) {
+            return Err(Error::UnknownOs(s.to_string()));
         }
+        Ok(Self(inner))
     }
 }
 
-impl Arch {
-    pub(crate) fn from_env() -> Result<Self, Error> {
-        Self::from_str(std::env::consts::ARCH)
+impl FromStr for Arch {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let inner = match s {
+            "x86" => target_lexicon::Architecture::X86_32(target_lexicon::X86_32Architecture::I686),
```

Comment:
Can you add a comment about the different `i?86` arches?

Reply:
Actually.. What do you want me to say? We just alias the i686 variant in parse and display for users.

Reply:
In this case just that it doesn't matter which of those we pick